src/machine.rs

```diff
@@ -883,15 +885,18 @@ impl OlmMachine {
     /// `predicate` is a closure that will be called for every known
     /// `InboundGroupSession`, which represents a room key. If the closure
     /// returns `true`, the `InboundGroupSession` will be included in the
-    /// export, otherwise it won't.
+    /// export; otherwise it won't.
+    ///
+    /// Returns a Promise containing a Result containing a String which is a
+    /// JSON-encoded array of ExportedRoomKey objects.
```

Comment:
(Unrelated to this specific PR, but) I wonder if we could make some kind of newtype that has phantom data of the right type, and when used with `wasm_bindgen` would be equivalent to a `Promise`. That way, we could have tighter checks for the return values of these functions, and we wouldn't need comments like this one, which has a high risk of getting outdated as soon as it lands.

Reply:
Also: is it a breaking change? I don't think so, because the previous code seemed to return a similar type, but maybe it did change :eyes: If it did, is it worth adding a note to the changelog?

Reply:
The return type is unchanged, just better documented :-)